spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/RowWriterFactory.scala

```diff
@@ -13,7 +13,7 @@ trait RowWriterFactory[T] {
   /** Creates a new `RowWriter` instance.
     * @param table target table the user wants to write into
     * @param columnNames columns selected by the user; the user might wish to write only a subset of columns */
-  def rowWriter(table: TableDef, columnNames: Seq[String]): RowWriter[T]
+  def rowWriter(table: TableDef, columnNames: Seq[String], checkColumns: CheckSetting): RowWriter[T]
```

Comment:
Thats good point, I'll extract this check out so that we can do column checking in the RDD implementation rather than in the rowWriter construction